larray/core/session.py

```diff
@@ -1590,7 +1603,9 @@ def _check_key_value(self, key, value):
         attr_def = getattr(cls, key, None)
         if attr_def is None:
             warnings.warn("'{}' is not declared in '{}'".format(key, self.__class__.__name__), stacklevel=2)
-        else:
+        elif isinstance(attr_def, (type, ArrayDef)):
+            if value is None:
```

Comment:
why do you need this special case for None?

Reply:
To not loose the order of the variables declared in the user defined class, I need to find a way to add the non-constant variables with a "temporary" value when calling the ``__init__`` method. For the moment, I add the non-constant variables with the value set to None in the ``__init__`` method until the user set a proper value later.

Reply:
> The usual trick is to create a global variable with an empty object as a unique flag:
```python
NOT_LOADED = object()
```
Thanks for the trick.

Reply:
> Also, it might (I have not thought this through) be useful to pass extra/optional arguments to a few functions including _check_key_value so that you know whether or not you are currently during __init__.

Since ``_check_key_value()`` is used in ``__setitem__`` and ``__setattr__`` methods and those methods do not accept extra arguments, I wonder if it is not better to add a flag as instance attribute to activate or deactivate the raising of an exception when a user try to set a value to a variable declared as constant?

Reply:
I don't know whether that's a good idea or not in this case, but you **can** define those methods with extra arguments (and use those extra arguments if you call those methods explicitly). It's just that the normal syntax/implicit way to call them will not **use** those extra arguments. See for example `LArray.__getitem__`